robber/matchers/numbers.py

```diff
@@ -34,10 +34,35 @@ def matches(self):
     def failure_message(self):
         return 'Expected %g to be within %g and %g' % (self.actual, self.expected, self.args[0])
 
+
+class Change(Base):
+    def __init__(self, callable, obj=None, *args):
+        self.callable = callable
+        self.obj = obj
+        self.args = args
+        self.message = None
+
+    def match(self):
+        return self
+
+    def by(self, amount=0):
+        changed = self.callable(self.obj) - self.obj
+
+        if changed != amount:
+            message = self.message or self.failure_message(self.obj, amount, changed)
```

Comment:
If we don't use lambda, is there a way to get a function name here? Then we could have a more meaningful failed message like:
```
Expect function `increase` to change 1 by 2, but was changed by 5?
```